domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -130,25 +134,65 @@ constructor(
         }
     }
 
-    override suspend fun takePicture() {
-        val imageDeferred = CompletableDeferred<ImageProxy>()
-
+    override suspend fun takePicture(
+        contentResolver: ContentResolver,
+        contentValues: ContentValues?,
+        takePictureCallback: TakePictureCallback
+    ) {
+        val outputFileOptions = OutputFileOptions.Builder(
+            contentResolver,
+            MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+            getEligibleContentValues(contentValues)
+        ).build()
         imageCaptureUseCase.takePicture(
+            outputFileOptions,
             defaultDispatcher.asExecutor(),
-            object : ImageCapture.OnImageCapturedCallback() {
-                override fun onCaptureSuccess(imageProxy: ImageProxy) {
-                    Log.d(TAG, "onCaptureSuccess")
-                    imageDeferred.complete(imageProxy)
+            object : ImageCapture.OnImageSavedCallback {
+                override fun onImageSaved(outputFileResults: ImageCapture.OutputFileResults) {
+                    Log.d(TAG, "Saved image to " + outputFileResults.savedUri)
+                    takePictureCallback.onPictureTaken(outputFileResults.savedUri)
                 }
 
                 override fun onError(exception: ImageCaptureException) {
-                    super.onError(exception)
-                    Log.d(TAG, "takePicture onError: $exception")
+                    Log.e(TAG, "Failed to save image.", exception)
+                    takePictureCallback.onError()
                 }
             }
         )
     }
 
+    private fun getEligibleContentValues(contentValues: ContentValues?): ContentValues {
+        val eligibleContentValues = ContentValues()
+        if (contentValues != null) {
+            for (key in contentValues.keySet()) {
+                val value = contentValues.get(key)
+                if (value == null) {
+                    continue;
+                } else if (value is String) {
+                    eligibleContentValues.put(key, value)
+                } else if (value is Int) {
+                    eligibleContentValues.put(key, value)
+                } else if (value is Boolean) {
+                    eligibleContentValues.put(key, value)
+                } else if (value is Long) {
+                    eligibleContentValues.put(key, value)
+                } else if (value is Float) {
+                    eligibleContentValues.put(key, value)
```

Comment:
This seems a bit weird, is this how it's supposed to be? If so that's fine

Reply:
Yeah, unfortunately there is no other way to copy ContentValues except iterating through every key-value pair, and this has to be done because put doesn't accept generic object type

Reply:
Could you just use [`ContentValues.putAll(ContentValues)`](https://developer.android.com/reference/android/content/ContentValues#putAll(android.content.ContentValues))?

Reply:
Actually this should no longer be needed if we're saving to a user-provided `Uri`.